### PR TITLE
BL-6138 Trace ID added to the main logs

### PR DESCRIPTION
**Pull request by Marcin-Duszynski**
### What are the relevant Jira tickets?
BL-6138

### Where should the reviewer start?
main.js

Trace id is readed from http header **x-mot-trace-id** field.

Log examples with trace id received from local MOTH:
```
{
   "name":"NotSet",
   "env":"NotSet",
   "functionName":"NotSet",
   "functionVersion":"NotSet",
   "level":"error",
   "context":{
...
   },
   "traceId":"1-1510949885-5a0f43fdcd531",
   "msg":"SMMT communication issue.",
   "time":"2017-11-17T20:18:05.877Z",
   "v":0
}
```

```
{
   "name":"NotSet",
   "env":"NotSet",
   "functionName":"NotSet",
   "functionVersion":"NotSet",
   "traceId":"1-123123-123123123123",
   "level":"info",
   "context":{
      "make":"BMW",
      "vin":"12345",
      "recall":{
         "success":true,
         "errors":[

         ],
         "description":"No Recall Outstanding",
         "status":"",
         "lastUpdate":"19022015"
      }
   },
   "msg":"Recall fetched successfully.",
   "time":"2017-11-20T12:21:40.244Z",
   "v":0
}
```

## Dev checklist:
- Is proper destination **branch** selected?
- Are proper **label** and **reviewers** selected?
- Has code been tested manually?
- Has code been covered by unit tests?
- Are unit test executed on Jenkins during pull request build?
- Do all unit tests pass?
- Is Jira ticket state updated?

## Code review checklist (for reviewers -> Copy to your comment)
- Is proper destination **branch** selected?
- Are proper **label** and **reviewers** selected?
- Do you understand task goal and business context?
- Is documentation updated and understandable?
- Has code been covered by unit tests?
- Did you identify technical issues/debt and reported it to TA? 
- Does code meet SOLID principles?
- Does code meet DRY principle?
- Is code in better state than before (scout rule)?

### CODE REVIEW

**Comment by Marcin-Duszynski** [comment]
Logger is injected and all logs will now contain traceId field

**Comment by Marcin-Duszynski** [comment]
New commits deliver logger injection and from now all logs will contain trace ID field fetched from request header.

File: app/test/smmt/client.test.js [hunk 1]
```diff
@@ -1,11 +1,39 @@
...
 
 const smmtClientFactory = require('../../src/smmt/client');
+const realLogger = require('../../src/logger/createLogger').create();
 
+const fakeLogger = sinon.stub(realLogger);
 chai.should();
 
+describe('When SMMT client is created', () => {
+  describe('and no parameters are provided to the create method', () => {
+    it('then error is return', () => {
```
Comment (ryszardm-kainos) [inline app/test/smmt/client.test.js]: return`ed`

File: app/test/smmt/client.test.js [hunk 2]
```diff
@@ -1,11 +1,39 @@
...
+      const smmtClient = smmtClientFactory.create();
+
+      smmtClient.should.be.a('Error');
+      smmtClient.should.have.property('message').eql('SMMT client arguments are missing.');
+    });
+  });
+
+  describe('and logger instance is not provided to the create method', () => {
+    it('then error is return', () => {
```
Comment (ryszardm-kainos) [inline app/test/smmt/client.test.js]: up

**Review by ryszardm-kainos — APPROVED** [review]

**Pull request merged**